### DISCUSSION TITLE
ModuleNotFoundError: No module named 'seaborn'

### DISCUSSION THREAD

**Discussion by jlsenaskcc** [cubewise-code/tm1py#755] [Q&A] [answered]
Hi,
I'm trying to run tm1py optimuspy connecting to a planning analytics server on  cloud and got this error messages:

Start quote:
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Input In [1], in <cell line: 14>()
     11 from TM1py import TM1Service
     12 from mdxpy import MdxBuilder, Member, MdxHierarchySet
---> 14 from executors import ExecutionMode, OriginalOrderExecutor, MainExecutor
     15 from results import OptimusResult
     17 APP_NAME = "optimuspy"

File ~\OneDrive - Kimberly-Clark\Documents\Files (UserDrive)\proyectos\python\optimus-py-master\executors.py:10, in <module>
      6 from typing import List, Dict
      8 from TM1py import TM1Service, Process
---> 10 from results import PermutationResult
     13 def swap(order: list, i1, i2) -> List[str]:
     14     seq = order[:]

File ~\OneDrive - Kimberly-Clark\Documents\Files (UserDrive)\proyectos\python\optimus-py-master\results.py:8, in <module>
      5 from pathlib import WindowsPath
      6 from typing import List, Union
----> 8 import seaborn as sns
     10 sns.set_theme()
     11 import matplotlib.pyplot as plt

ModuleNotFoundError: No module named 'seaborn'

End quote.

Any ideas? Am I missing something?
Many tnaks.

**Comment by rclapp** [comment, marked as answer]
Yes, you need to install seaborn lib. Run pip install seaborn

Sent from my mobile device

On Jun 30, 2022 7:22 PM, jlsenaskcc ***@***.***> wrote:

Hi,
I'm trying to run tm1py optimuspy connecting to a planning analytics server on cloud and got this error messages:

Start quote:

ModuleNotFoundError Traceback (most recent call last)
Input In [1], in <cell line: 14>()
11 from TM1py import TM1Service
12 from mdxpy import MdxBuilder, Member, MdxHierarchySet
---> 14 from executors import ExecutionMode, OriginalOrderExecutor, MainExecutor
15 from results import OptimusResult
17 APP_NAME = "optimuspy"

File ~\OneDrive - Kimberly-Clark\Documents\Files (UserDrive)\proyectos\python\optimus-py-master\executors.py:10, in
6 from typing import List, Dict
8 from TM1py import TM1Service, Process
---> 10 from results import PermutationResult
13 def swap(order: list, i1, i2) -> List[str]:
14 seq = order[:]

File ~\OneDrive - Kimberly-Clark\Documents\Files (UserDrive)\proyectos\python\optimus-py-master\results.py:8, in
5 from pathlib import WindowsPath
6 from typing import List, Union
----> 8 import seaborn as sns
10 sns.set_theme()
11 import matplotlib.pyplot as plt

ModuleNotFoundError: No module named 'seaborn'

End quote.

Any ideas? Am I missing something?
Many tnaks.

-

**Comment by MariusWirtz** [comment]
I included seaborn in the readme.md
https://github.com/cubewise-code/optimus-py

Thanks for raising the issue @jlsenaskcc
